Gazetteer search

annarieger opened the pull request:
This PR introduces a gazetteer search based on OSM dataset.

Many thanks to @KaiVolland for the great preliminary work!

Please review /cc @KaiVolland, @weskamm, @buehner, @marcjansen.

marcjansen commented on src/view/form/field/GazetteerCombo.js:
```diff
@@ -0,0 +1,116 @@
+        me.nav = Ext.create('Ext.util.KeyNav', me.el, {
+            esc: function(){
+                me.clearValue();
+            },
```
You shouldn't need the anonymous function:

```js
me.nav = Ext.create('Ext.util.KeyNav', me.el, {
  esc: me.clearValue,
  scope: me
});
```

marcjansen commented on src/view/form/field/GazetteerCombo.js:
```diff
@@ -0,0 +1,116 @@
+                    duration: 250,
+                    callback: function(){
+                        gazetteerGrid.hide();
+                    }
```
The anonymous function here can probably as well be removed (if we can also provide the scope.)

marcjansen commented on src/view/grid/GazetteerGrid.js:
```diff
@@ -0,0 +1,269 @@
+            hideToolTooltip: 'Gazetteer verbergen',
+            limitCboxLabel: 'Auf den sichtbaren Kartenbereich einschränken',
+            refreshBtnTooltip: 'Aktualisieren',
+            directionBtnTooltip: 'Hinweise zu Gazetteer'
```
Do we have German everywhere else?

annarieger replied:
German tooltips and strings will be used e.g. in all buttons and other components such as [`LayerSlider`](https://github.com/terrestris/BasiGX/blob/master/src/view/container/LayerSlider.js) or [`NominatimSearch`](https://github.com/terrestris/BasiGX/blob/master/src/view/container/LayerSlider.js).
However, some components are (still?) in English (e.g. [`RedliningStyler`](https://github.com/terrestris/BasiGX/blob/master/src/view/container/RedlineStyler.js)). So there is a bit inconsistency between BasiGX classes. Probably we should discuss the language/translation concept later once again.

marcjansen commented on src/view/grid/GazetteerGrid.js:
```diff
@@ -0,0 +1,269 @@
+                    'Beachten Sie, dass einige Suchergebnisse außerhalb des ' +
+                    'sichtbaren Kartenbereichs liegen können, falls die ' +
+                    'Option "Auf den sichtbaren Kartenbereich einschränken" ' +
+                    'abgewählt ist.'
```
Should this be in a cfg or in a viewModel?

marcjansen commented on src/view/grid/GazetteerGrid.js:
```diff
@@ -0,0 +1,269 @@
+     */
+    onBoxReady: function(){
+        var me = this;
+        if(!me.map){
```
Shouldn't we use the `getMap()` method here? Can we be sure that `{config: {map: null}}` will be stored inside of the property `map`?

marcjansen commented on src/view/grid/GazetteerGrid.js:
```diff
@@ -0,0 +1,269 @@
+    },
+
+    tools:[{
+        type:'minimize',
```
missing a space

marcjansen commented:
I was quite picky this time, @annarieger. I have added a couple of comments which you may want to address. Nothing to hard, though. Please merge once done. Thanks!

annarieger commented:
Many thanks for the review @marcjansen! I've addressed all your remarks except the german translations - this issue should be probably discussed separately later.

Merging now..
